### ISSUE TITLE
Solution for permissions

### ISSUE THREAD

**Issue by jdkasten** [letsencrypt/letsencrypt#552]
make_or_verify_dir was initially written to avoid privilege escalation attacks.  It checks that the directory is owned by the current user.  Since files are overwritten and files are transferred into protected directories as root, it was important that those files were not edited by a lower privileged user.  Requiring blanket permissions was an easy way to avoid the vulnerabilities.  

Example vuln:  Backup configuration files for Apache will automatically be transferred into the configuration if Let's Encrypt fails... or rather if certain backup files exist.  Apache configuration files are Turing complete.

Now that the project has transitioned into a much more robust and feature heavy application.  This limitation can be confusing and annoying.

We need a finer-grained approach that maintains the protections where necessary, but also avoids unnecessary restrictions.

**Comment by schoen** [comment]
Related to #420, which on examination addresses a different topic (partly making sure that we don't create files and directories with incorrectly lax permissions, for example by using `os.mkdir` instead of `make_or_verify_dir`).

**Comment by bkleef** [comment]
Especially important when using Docker volumes on OS X. You get totally stuck like this:
```
bash-4.3# whoami
root
```
```
bash-4.3# mount | grep letsencrypt
none on /etc/letsencrypt type vboxsf (rw,nodev,relatime)
none on /var/lib/letsencrypt type vboxsf (rw,nodev,relatime)
```
```
bash-4.3# ls -l /etc/letsencrypt | awk '{k=0;for(i=0;i<=8;i++)k+=((substr($1,i+2,1)~/[rwx]/)*2^(8-i));if(k)printf("%0o ",k);print}'
total 0
644 -rw-r--r--    1 1000     50               0 Aug  4 14:13 cli.ini
```
```
bash-4.3# chmod 755 /etc/letsencrypt
```
```
bash-4.3# ls -l /etc/letsencrypt | awk '{k=0;for(i=0;i<=8;i++)k+=((substr($1,i+2,1)~/[rwx]/)*2^(8-i));if(k)printf("%0o ",k);print}'
total 0
644 -rw-r--r--    1 1000     50               0 Aug  4 14:13 cli.ini
```

**Comment by jdkasten** [comment]
This was discussed and resolved on IRC.  The client expects certain permissions and ownership of directories.  It expects the owner is the user running the script and it expects that permissions are what they were, when they were initialized.  If the directories do not exist at the time of initialization the client will automatically create them.

It is not recommended that you modify these before or after installation at this time.

**Comment by jdkasten** [comment]
Fixed in #778
